spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/LocalPredictor.scala

```diff
@@ -0,0 +1,182 @@
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Engine, MklBlas}
+
+
+import scala.reflect.ClassTag
+
+object LocalPredictor {
+  def getAndClearWeightBias[T: ClassTag](parameters: (Array[Tensor[T]], Array[Tensor[T]]))
+                                        (implicit ev: TensorNumeric[T]): Array[Tensor[T]] = {
+    var i = 0
+    val weightsBias = new Array[Tensor[T]](parameters._1.length)
+    while (i < parameters._1.length) {
+      if (parameters._1(i) != null) {
+        val wb = parameters._1(i)
+        weightsBias(i) = Tensor[T](Storage(wb.storage().array()),
+          wb.storageOffset(), wb.size(), wb.stride())
+      }
+      i += 1
+    }
+    i = 0
+    while (i < parameters._1.length) {
+      if (parameters._1(i) != null) {
+        parameters._1(i).set()
+      }
+      if (parameters._2(i) != null) {
+        parameters._2(i).set()
+      }
+      i += 1
+    }
+    weightsBias
+  }
+
+  def apply[T: ClassTag](model: Module[T], weightsBias: Array[Tensor[T]])
+                        (implicit ev: TensorNumeric[T]): LocalPredictor[T] = {
+    new LocalPredictor[T](model, weightsBias)
+  }
+}
+
+class LocalPredictor[T: ClassTag] private[optim](model: Module[T], weightsBias: Array[Tensor[T]])
+                                                (implicit ev: TensorNumeric[T])
+  extends Serializable {
+
+  val logger = LocalValidator.logger
+  private val coreNumber = Engine.coreNumber()
+
+  private val subModelNumber = Engine.getEngineType match {
+    case MklBlas => coreNumber
+    case _ => throw new IllegalArgumentException
+  }
+
+  private val batchSize = 8
+
+  def predictClass(dataSet: Array[Sample[T]]): Array[Int] = {
+    val result = predict(dataSet)
+    result.map(output => {
+      val _output = output.toTensor[T]
+      require(_output.dim() == 1, s"Predictor.predictClass:" +
+        s"Only support one sample has one lable, but got ${_output.dim()} label")
+      ev.toType[Int](_output.max(1)._2.valueAt(1))
+    })
+  }
+
+  def predictClass(dataSet: LocalDataSet[Sample[T]]): Array[Int] = {
+    val result = predict(dataSet)
+    result.map(output => {
+      val _output = output.toTensor[T]
+      require(_output.dim() == 1, s"Predictor.predictClass:" +
+        s"Only support one sample has one lable, but got ${_output.dim()} label")
+      ev.toType[Int](_output.max(1)._2.valueAt(1))
+    })
+  }
+
+  def predict(dataSet: LocalDataSet[Sample[T]]): Array[Activity] = {
+    val dataset = dataSet.transform(SampleToBatch[T](
+      batchSize = batchSize, None, None, None,
+      partitionNum = Some(1))).asInstanceOf[LocalDataSet[MiniBatch[T]]]
+    val dataIter = dataset.data(train = false)
+
+    val workingModels = (1 to subModelNumber).map(_ => {
+      val submodel = model.cloneModule().evaluate()
+      putWeightBias(weightsBias, submodel)
+      submodel
+    }).toArray
```

Comment:
weightsBias = getAndClearWeightBias(model)
workingModels[0] = model.evaluate()
(1 to subModelNumber-1).map(i => {
   workingModels[i] = model.cloneModule()
   putWeightBias(weightsBias, workingModels[i])
}
putWeightBias(weightsBias, workingModels[0])